enterprise/server/util/vsock/vsock.go

```diff
@@ -140,10 +140,12 @@ func dialHostToGuest(ctx context.Context, socketPath string, port uint32) (net.C
 	if n != len(fcConnectString) {
 		return nil, status.InternalErrorf("HostDial failed: wrote %d bytes, expected %d", n, len(fcConnectString))
 	}
+	conn.SetReadDeadline(time.Now().Add(1 * time.Second))
```

Comment:
```suggestion
	// Firecracker should normally be listening on v.sock as soon as
	// the VM is started, but occasionally we might call this function
	// before firecracker starts listening. Set a short timeout so that
	// we catch this early and retry instead of timing out.
	conn.SetReadDeadline(time.Now().Add(1 * time.Second))
```